src/plugins/mf_priority.cpp

```diff
@@ -234,6 +237,37 @@ static int conf_update_cb (flux_plugin_t *p,
     if (queue_weight != -1)
         priority_weights["queue"] = queue_weight;
 
+    if (queue_priorities) {
+        const char *key;
+        json_t *value;
+
+        json_object_foreach (queue_priorities, key, value) {
+            if (!key || !json_is_integer (value)) {
+                flux_log_error (h, "mf_priority: invalid data in queue-priorities");
+                continue;
+            }
+
+            const std::string queue_name (key);
+            int priority = json_integer_value (value);
+
+            auto it = queues.find (queue_name);
+            if (it != queues.end ()) {
+                // update the priority for the existing queue
+                it->second.priority = priority;
+            } else {
+                // queue does not exist; create a new queue and insert it
+                // and set the limits to their default values defined in
+                // the flux-accounting DB
+                Queue new_queue;
+                new_queue.min_nodes_per_job = 1;
+                new_queue.max_nodes_per_job = 1;
+                new_queue.max_time_per_job = 60;
+                new_queue.priority = priority;
+                queues[queue_name] = new_queue;
```

Comment:
Can you remind me if it is an error if a queue exists in the Flux broker config, but does not exist in the flux-accounting database?

I'm wondering if these defaults make sense if, say, an admin wanted to configure a temporary queue with a high priority. The defaults of max nodes (1)  and max time (60m) would not necessarily make sense in this case...

Reply:
Good question. I don't think it is an error - I think the plugin will just return `UNKNOWN_QUEUE`, meaning it can't find any information about the queue, so it will just apply a priority of `0` for that queue. Perhaps I should just leave out the initialization of max nodes and max time when adding this queue to the map?

Reply:
That's what I was thinking (just leave out initialization of maximums here). However, we should ask @kkier and @ryanday36 if there is a use case for adding queue priorities for queues  not in the accounting db.

Reply:
Cool. FWIW, those maximums are not currently enforced anywhere in the plugin (yet). But probably good to get @ryanday36 and @kkier's thoughts on it too.

Reply:
I can confirm that it is not an error for queues to exist in the broker config but not the accounting database. I've generally only added queues to the accounting database when I have a specific need to do so (i.e. I need to limit access to the queue to specific users). The use case that I can see for queue priorities without the accounting db would be pre-production systems where we're not otherwise worried about banks / fairshare / per-user limits / etc. It might be useful to be able to configure what the accounting plugin is enforcing a bit more (e.g. enforce queue limits, but don't require banks or calculate fairshare). It's pretty easy to set up a dummy-ish bank (`guests`) and add everyone to it though, which is what we're usually actually doing.